packages/minimongo/sorter.js

```diff
@@ -140,9 +140,9 @@ export default class Sorter {
       let branches = expandArraysInBranches(spec.lookup(doc), true);
 
       // If there are no values for a key (eg, key goes to an empty array),
-      // pretend we found one null value.
+      // pretend we found one undefined value.
       if (!branches.length) {
-        branches = [{value: null}];
+        branches = [{value: undefined}];
```

Comment:
null is sorted higher than undefined by mongo. But an empty array should be considered the same as undefined (ref: https://docs.mongodb.com/manual/reference/bson-type-comparison-order/#arrays)